src/main/java/com/google/step/finscholar/data/Scholarship.java

```diff
@@ -0,0 +1,308 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.step.finscholar.data;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+/** The Scholarship Class. */
+public class Scholarship {
+
+  private final String scholarshipName;
+  private final UUID scholarshipUUID;
+  private final List<UUID> schoolsList;
+  // For amountPerYear, I use String instead of integer because lots of expression for amount is 
+  // irregular, such as "full tuition with extra $1000 bonus", "full tuition with room & board" etc..
+  private Optional<String> amountPerYear;
+  private Optional<String> applicationProcess;
+  private Optional<Integer> numberOfYears;
+  private List<String> ethnicityRaceRequirements;
+  private List<String> genderRequirements;
+  private List<String> nationalOriginRequirements;
+  private List<String> locationRequirements;
+  private List<String> financialRequirements;
+  private List<String> academicRequirements;
+  private List<String> otherRequirements;
+  // Introduction can be any background knowledge of the scholarship.
+  private Optional<String> introduction;
+  private Optional<Boolean> isRenewable;
+  private final String URL;
+
+  public Scholarship(String name, UUID uuid, List<UUID> schoolsList, String URL) {
+    this.scholarshipName = name;
+    this.scholarshipUUID = uuid;
+    this.schoolsList = schoolsList;
+    this.URL = URL;
+    this.amountPerYear = Optional.empty();
+    this.applicationProcess = Optional.empty();
+    this.numberOfYears = Optional.empty();
+    this.ethnicityRaceRequirements = List.of();
+    this.genderRequirements = List.of();
+    this.nationalOriginRequirements = List.of();
+    this.locationRequirements = List.of();
+    this.financialRequirements = List.of();
+    this.otherRequirements = List.of();
+    this.isRenewable = Optional.empty();
+  }
+
+  /** 
+   * The getter of academic requirments list. 
+   * @return The academic requirements list for the scholarship. 
+   */
+  public List<String> getAcademicRequirements() {
+    return academicRequirements;
+  }
+
+  /**
+   * The getter of applicationProcess.
+   * @return The string overview of application process.
+   */
+  public Optional<String> getApplicationProcess() {
+    return this.applicationProcess;
+  }
+
+  /** 
+   * The getter of the amount of money sponsored per year. 
+   * @return The amount given to one individual per year.
+   */
+  public Optional<String> getAmountPerYear() {
```

Comment:
I'd say the main thing is to return this.instanceVariable for all of the private instance variable getters. Otherwise, it looks good to me.

Reply:
Yep I changed them in the newest commit.